src/server/zset_family.cc

```diff
@@ -2396,37 +2396,61 @@ OpResult<MScoreResponse> ZSetFamily::ZGetMembers(CmdArgList args, ConnectionCont
 void ZSetFamily::GeoAdd(CmdArgList args, ConnectionContext* cntx) {
   string_view key = ArgS(args, 0);
 
-  // TODO: to handle options and multiple elements
   ZParams zparams;
+  size_t i = 1;
+  for (; i < args.size() - 1; ++i) {
+    ToUpper(&args[i]);
 
-  string_view longitude = ArgS(args, 1);
-  string_view latitude = ArgS(args, 2);
-  string_view member = ArgS(args, 3);
+    string_view cur_arg = ArgS(args, i);
 
-  // TODO: to remove this check once the TODO above is handled.
-  if (args.size() != 4) {
-    return (*cntx)->SendError(kSyntaxErr);
+    if (cur_arg == "XX") {
+      zparams.flags |= ZADD_IN_XX;  // update only
+    } else if (cur_arg == "NX") {
+      zparams.flags |= ZADD_IN_NX;  // add new only.
+    } else if (cur_arg == "CH") {
+      zparams.ch = true;
+    } else {
+      break;
+    }
+  }
+
+  if ((args.size() - i) % 3 != 0) {
+    (*cntx)->SendError(kSyntaxErr);
+    return;
   }
 
-  // TODO: the code handles only a single tripple of long,lat,member.
-  //       it has to be extended to handle multiple elements.
-  pair<double, double> longlat;
-  for (int i = 0; i < 1; i++) {
+  unsigned insert_mask = zparams.flags & (ZADD_IN_NX | ZADD_IN_XX);
```

Comment:
I would drop the `insert_mask` var and change the `if` below to:
```cpp
if ((zparams.flags & ZADD_IN_NX) && (zparams.flags & ZADD_IN_XX))
```
Or do you think it's less readable?

Reply:
Like the comment above, took the `ZADD` implementation for reference. I agree with you on readability but was trying to keep the code aligned to the `ZADD` implementation.